### PR TITLE
[11.0][IMP] stock: fix company on quants for customers/supplier location

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
v11: https://github.com/OCA/OpenUpgrade/blob/11.0/addons/stock/models/stock_quant.py#L31 --> in practice this is null as the company is null for customer locations, at least it should.

v10:  https://github.com/OCA/OpenUpgrade/blob/10.0/addons/stock/models/stock_quant.py#L61 --> in practice this is the company that does the delivery in case of cusotmer locations

cc @MiquelRForgeFlow

### CODE REVIEW

**Comment by pedrobaeza** [comment]
This is not correct at all, as you are assuming the destination location doesn't have an assigned company. You should synchronize the location company with the quant company.

**Comment by AaronHForgeFlow** [comment]
In case the location has a company that code does not change the company in the quant. Both are already synchronized when there is a company in the location.

The only case when both do not match is when the company is empty in the location

**Comment by AaronHForgeFlow** [comment]
@pedrobaeza  does it look better now?

**Comment by pedrobaeza** [comment]
No. I still insists you can unconditionally empty the company. You should just empty it if they customer location has empty company.

**Comment by AaronHForgeFlow** [comment]
Ok, I included that condition, I think this is safe to be included in OU now.

File: addons/stock/migrations/11.0.1.1/pre-migration.py [hunk 1]
```diff
@@ -22,6 +22,26 @@ def delete_quants_for_consumable(env):
...
+    """
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE stock_quant sq
+        SET company_id = sl.company_id
+        FROM stock_location sl
+        WHERE sl.id = sq.location_id
+            AND sl.usage in ('customer', 'supplier')
+            AND sq.company_id IS NOT NULL
+            AND sl.company_id IS NULL
```
Comment (pedrobaeza) [inline addons/stock/migrations/11.0.1.1/pre-migration.py]:
```suggestion
        UPDATE stock_quant sq
        SET company_id = sl.company_id
        FROM stock_location sl
        WHERE sl.id = sq.location_id
            AND sl.usage in ('customer', 'supplier')
            AND sq.company_id != sl.company_id
```

Comment (AaronHForgeFlow) [reply addons/stock/migrations/11.0.1.1/pre-migration.py]: I finally got the point :facepalm:

**Pull request merged**